app/models/manager_refresh/inventory_collection/data_storage.rb

```diff
@@ -31,7 +32,7 @@ def initialize(inventory_collection, secondary_refs)
       end
 
       def <<(inventory_object)
-        unless primary_index.find(inventory_object.manager_uuid)
+        if inventory_object.manager_uuid.present? && !primary_index.find(inventory_object.manager_uuid)
```

Comment:
Why would the manager_uuid not be present now?

Reply:
if we e.g. build with :ems_ref => nil

Reply:
Is that something we want to allow?

Reply:
right, this is compatible with current behavior, just moved. It will just result to nil